src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -652,78 +674,57 @@ Result SpatialJoinAlgorithms::BoundingBoxAlgorithm() {
-      entry.geoPoint = geopoint.value();
-      // create a box with a side length of at most 1mm to approximate the point
-      bbox = Box(Point(geopoint.value().getLng(), geopoint.value().getLat()),
-                 Point(geopoint.value().getLng() + 0.00000001,
-                       geopoint.value().getLat() + 0.00000001));
+    // add every box together with the additional information into the rtree
+    std::optional<rtreeEntry> entry =
+        getRtreeEntry(smallerResult, i, smallerResJoinCol);
+    if (!entry) {
+      // nothing to do. When parsing a point or an area fails, a warning
+      // message gets printed at another place and the point/area just gets
+      // skipped
+      continue;
     }
-
-    // add every box together with the row number into the rtree
-    rtree.insert(std::make_pair(std::move(bbox), std::move(entry)));
+    rtree.insert(std::make_pair(entry.value().boundingBox_.value(),
+                                std::move(entry.value())));
   }
 
   // query rtree with the other child
   std::vector<Value, ad_utility::AllocatorWithLimit<Value>> results{
       qec_->getAllocator()};
   for (size_t i = 0; i < otherResult->numRows(); i++) {
-    auto geopoint = getPoint(otherResult, i, otherResJoinCol);
-    std::vector<Box> bbox;
-    rtreeEntry entry{i, std::nullopt, std::nullopt, std::nullopt};
-    entry.row = i;
-
-    if (!geopoint) {
-      AnyGeometry geometry;
-      if (!getAnyGeometry(otherResult, i, otherResJoinCol, geometry)) {
-        // nothing to do. When parsing a point or an area fails, a warning
-        // message gets printed at another place and the point/area just gets
-        // skipped
-        continue;
-      }
-      auto areaBox = boost::apply_visitor(BoundingBoxVisitor(), geometry);
-      entry.geometry = std::move(geometry);
-      auto midpoint = calculateMidpointOfBox(areaBox);
-      bbox = computeBoundingBox(
-          midpoint,
-          getMaxDistFromMidpointToAnyPointInsideTheBox(areaBox, midpoint));
-
+    std::optional<rtreeEntry> entry =
+        getRtreeEntry(otherResult, i, otherResJoinCol);
+    if (!entry) {
+      // nothing to do. When parsing a point or an area fails, a warning
+      // message gets printed at another place and the point/area just gets
+      // skipped
+      continue;
+    }
+    std::vector<Box> queryBox;
+    if (!entry.value().geoPoint_) {
+      auto midpoint =
+          calculateMidpointOfBox(entry.value().boundingBox_.value());
+      queryBox = computeBoundingBox(
+          midpoint, getMaxDistFromMidpointToAnyPointInsideTheBox(
+                        entry.value().boundingBox_.value(), midpoint));
     } else {
-      entry.geoPoint = geopoint;
-      bbox = computeBoundingBox(
-          Point(geopoint.value().getLng(), geopoint.value().getLat()));
+      queryBox =
+          computeBoundingBox(Point(entry.value().geoPoint_.value().getLng(),
+                                   entry.value().geoPoint_.value().getLat()));
```

Comment:
This whole thing (get the vector<BoundingBox> for a query with the added distance) should also be a separate function.